haystack/reader/base.py

```diff
@@ -50,7 +50,7 @@ def run(self, query: str, documents: List[Document], top_k_reader: Optional[int]
         if documents:
             results = self.predict(query=query, documents=documents, top_k=top_k_reader)
         else:
-            results = {"answers": []}
+            results = {"answers": [], "query": query}
```

Comment:
Why are we adding this here? Is that potentially a breaking change?

Reply:
This change makes the response compliant with the results format for the case when no answers are found. It should not be a breaking change.

Reply:
Isn't is rather for the case when the retriever does not pass any docs to the reader?

Reply:
Yes, it returns a response compliant with the results format.

Reply:
The same is done in the Finder [here](https://github.com/deepset-ai/haystack/blob/4152ad8426cf5132cabf0792548c7c0c4affbae9/haystack/finder.py#L75).